DependencyInjection/DoctrineMongoDBExtension.php

```diff
@@ -427,6 +395,40 @@ private function loadMessengerServices(ContainerBuilder $container): void
         $loader->load('messenger.xml');
     }
 
+    /** @param array<string, mixed> $config */
+    private function loadEntityValueResolverServices(ContainerBuilder $container, FileLoader $loader, array $config): void
+    {
+        // available in Symfony 6.2 and higher
+        if (! class_exists(EntityValueResolver::class)) {
+            return;
+        }
+
+        $loader->load('value_resolver.xml');
+
+        if (! class_exists(ExpressionLanguage::class)) {
+            $container->removeDefinition('doctrine_mongodb.odm.entity_value_resolver.expression_language');
+        }
+
+        $controllerResolverDefaults = [];
+
+        if (! $config['controller_resolver']['enabled']) {
+            $controllerResolverDefaults['disabled'] = true;
+        }
+
+        if (! $config['controller_resolver']['auto_mapping']) {
+            $controllerResolverDefaults['mapping'] = [];
+        }
+
+        if ($controllerResolverDefaults === []) {
+            return;
+        }
+
+        $mapEntityDefinition = $container->getDefinition('doctrine_mongodb.odm.entity_value_resolver.map_entity');
+
+        $mapEntityDefinition->setArgument('$mapping', $controllerResolverDefaults['mapping'] ?? null);
+        $mapEntityDefinition->setArgument('$disabled', $controllerResolverDefaults['disabled'] ?? false);
```

Comment:
@alcaeus I've registered the service `doctrine_mongodb.odm.entity_value_resolver.map_entity` previously and this is another way to only set some parameters, not sure if this is recommended though since if the name of the argument is changed, this will fail.